src/Helpers/Utils.php

```diff
@@ -141,6 +152,14 @@ public static function containsSerializedDataRegex($data): bool
             return false;
         }
 
+        $data = self::recursiveUrlDecode($data);
+
+        /**
+         * This regular expression removes any special character that is not:
+         * a Letter (a-zA-Z), number (0-9), or any of the characters {}, :, ;, ", ', ., [, ], (, ), ,
+         */
+        $data = preg_replace('/[^a-zA-Z0-9:{};"\'\.\[\]\(\),]/', '', $data);
```

Comment:
Looks like we don't need to escape every character (from php storm below)
<img width="664" alt="Screenshot 2025-01-07 at 9 27 22 AM" src="https://github.com/user-attachments/assets/e11bc4df-8f00-4ccb-b5f9-d26d80bdf902" />